ocs_ci/ocs/resources/bucketclass.py

```diff
@@ -0,0 +1,123 @@
+
+from ocs_ci.ocs.ocp import OCP
+import logging
+from ocs_ci.framework import config
+from tests import helpers
+
+log = logging.getLogger(__name__)
+
+
+class BucketClass():
+    """
+    A class that represents BucketClass objects
+
+    """
+    def __init__(self, name, backingstores, placement):
+        self.name = name
+        self.backingstores = backingstores
+        self.placement = placement
+
+    # TODO: verify health of bucketclass
+
+    def delete(self):
+        log.info(f'Cleaning up bucket class {self.name}')
+
+        OCP(
+            namespace=config.ENV_DATA['cluster_namespace']
+        ).exec_oc_cmd(
+            command=f'delete bucketclass {self.name}',
+            out_yaml_format=False
+        )
+
+        log.info(
+            f"Verifying whether bucket class {self.name} exists after deletion"
+        )
+        # Todo: implement deletion assertion
+
+
+def bucket_class_factory(request, mcg_obj, backingstore_factory):
+    """
+    Create a bucket class factory. Calling this fixture creates a new custom bucket class.
+    For a custom backingstore(s), provide the 'backingstore_dict' parameter.
+
+    Args:
+        request (object): Pytest built-in fixture
+        mcg_obj (MCG): An MCG object containing the MCG S3 connection credentials
+        backingstore_factory: Factory for backing store creation
+
+    """
+    interfaces = {
+        'oc': mcg_obj.oc_create_bucketclass,
+        'cli': mcg_obj.cli_create_bucketclass
+    }
+    created_bucket_classes = []
+
+    def _create_bucket_class(bucket_class_dict):
+        """
+        Creates and deletes all bucket classes that were created as part of the test
+
+        Args:
+            bucket_class_dict (dict): Dictionary containing the description of the bucket class.
+                possible keys and values are:
+                - interface (str): The interface to use for creation of buckets.
+                    OC | CLI
+                - backingstore_dict (dict/list): Either existing list of backingstores or
+                    A dictionary compatible with the backing store factory requirements.
+                - placement (str): The Placement policy for this bucket class.
+                    Spread | Mirror
+                if no key is provided default values will apply.
+
+        Returns:
+            list: A Bucket Class object.
+
+        """
+        if 'interface' in bucket_class_dict:
+            interface = bucket_class_dict['interface']
+            if interface.lower() not in interfaces.keys():
+                raise RuntimeError(
+                    f'Invalid interface type received: {interface}. '
+                    f'available types: {", ".join(interfaces)}'
+                )
+            else:
+                interface = bucket_class_dict['interface']
+        else:
+            interface = 'OC'
+        if 'backingstore_dict' in bucket_class_dict:
+            if isinstance(bucket_class_dict['backingstore_dict'], dict):
```

Comment:
it's a bit confusing that `bucket_class_dict` doesn't need to be a dict. Please consider again accepting only one type to the method

Reply:
this is actually the `backing_store_dict` and as mentioned above i will change it in the next commit to be a dictionary or nothing.